### PR TITLE
Support for piecewise split of Job result

### PR DESCRIPTION
**Pull request by mdonaka**
# Summary

split job result

# Purpose

Successfully save to DB and return to Client when job result is huge.

# Contents

### CC

- split job result to piece on writeDB
    - add piece_id to meta information
    - Default split by 1 MB each
- combine model param pieces on readModelParam

### MC

- support getComputationResult to stream
- add piece_id to insertModelParams

### proto

- add stream to get_computation_result of lib2MC
- add piece_id to send_model_param of lib2MC

# Testing Methods Performed
- CI
- Send a request to get a join table job with larger data

### CODE REVIEW

**Comment by arukuka** [comment]
@mdonaka Thank you four your contribution.  
I would like to check new CI status.  
Please rebase or merge latest develop because #6 is merged.

**Comment by mdonaka** [comment]
Merged develop and passed CI done.
@arukuka Thanks for the CI fix!

File: src/ComputationContainer/Test/IntegrationTest/ComputationToDbGateTest.hpp [hunk 1]
```diff
@@ -1,14 +1,18 @@
 #include "gtest/gtest.h"
+#include <map>
 #include "nlohmann/json.hpp"
 #include "Client/ComputationToDbGate/Client.hpp"
```
Comment (arukuka) [inline src/ComputationContainer/Test/IntegrationTest/ComputationToDbGateTest.hpp:4]: It is preferable to `include` the C++ standard library headers first.

Please refer following link:  
https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes

```suggestion
#include <map>

#include "gtest/gtest.h"
#include "nlohmann/json.hpp"

#include "Client/ComputationToDbGate/Client.hpp"
```

**Review by arukuka — CHANGES_REQUESTED** [review]
Thank you for your contirbution!  
There are some points which concern me, please check it.

File: src/ComputationContainer/Client/ComputationToDbGate/README.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+}
+```
+
+#### join table
+resultはJson文字列
+```Json
+{
+  "job_uuid": "10b6bfda-8776-4d8c-ad9a-b40fa277ff37",
+  "status": 5,
+  "result": "\"{\"schema\":[\"s1\",\"s2\"],\"table\":[[\"1\",\"2\"],[\"3\",\"4\"]]}\""}
```
[the remark this answers is not in the document]
Comment (mdonaka) [reply src/ComputationContainer/Client/ComputationToDbGate/README.md]: Thanks for noticing!

File: src/ComputationContainer/Test/IntegrationTest/ComputationToDbGateTest.hpp (hunk 1 above)
Comment (mdonaka) [reply src/ComputationContainer/Test/IntegrationTest/ComputationToDbGateTest.hpp]: I forgot that sorting is false in format. Thanks!

**Review by arukuka — APPROVED** [review]

File: src/ManageContainer/Client/ManageToDbGate/Client.go [hunk 3]
```diff
@@ -141,10 +146,20 @@ func (c Client) InsertShares(dataID string, schema []string, pieceID int32, shar
...
+	} else if bucket == "result" {
+		idName = "job_uuid"
+	} else {
+		err := fmt.Errorf("'bucket' must be `share` or `result`. %s is not found.", bucket)
+		return false, err
+	}
+
+	ls.Lock(ID)
+	defer ls.Unlock(ID)
+	query := fmt.Sprintf("SELECT x.meta, meta().id FROM `%s` x WHERE x.%s= '%s';", bucket, idName, ID)
```
Comment (come2ry) [inline src/ManageContainer/Client/ManageToDbGate/Client.go:162]: @mdonaka 
There's not much point in doing Escape this time, but you should use it to make sure you don't forget Escape when you really need it.

```suggestion
	query := fmt.Sprintf("SELECT x.meta, meta().id FROM `%s` x WHERE x.%s= '%s';", 
		bucket, utils.EscapeInjection(idName, utils.Where), utils.EscapeInjection(ID, utils.Where))
```

**Review by come2ry — CHANGES_REQUESTED** [review]

File: src/ManageContainer/Client/ManageToDbGate/Client.go [hunk 4]
```diff
@@ -322,55 +337,47 @@ func (c Client) getComputationResult(conn *grpc.ClientConn, jobUUID string) (*Co
...
-	defer ls.Unlock(jobUUID)
+func (c Client) insertModelParams(conn *grpc.ClientConn, jobUUID string, params string, pieceId int32) error {
+	// 重複チェック
+	exist, err := c.existPieceID(conn, "result", jobUUID, pieceId)
 	if err != nil {
 		return err
 	}
-	if cnt > 0 {
+	if exist {
 		return errors.New("重複データ登録エラー: " + jobUUID + "は既に登録されています．")
```
Comment (come2ry) [inline src/ManageContainer/Client/ManageToDbGate/Client.go:371]: @mdonaka 
It's more consistent this message in English.

```suggestion
		return errors.New("Duplicate error: " + jobUUID + " has already been registered.")
```

File: src/ManageContainer/Client/ManageToDbGate/Client.go (hunk 3 above)
Comment (mdonaka) [reply src/ManageContainer/Client/ManageToDbGate/Client.go]: Thanks for pointing that out!

File: src/ManageContainer/Client/ManageToDbGate/Client.go (hunk 4 above)
Comment (mdonaka) [reply src/ManageContainer/Client/ManageToDbGate/Client.go]: Thanks for Suggested! I committed.

**Review by come2ry — APPROVED** [review]

**Pull request merged**